packages/fx-core/src/plugins/resource/cicd/plugin.ts

```diff
@@ -0,0 +1,91 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+import { Context } from "@microsoft/teamsfx-api/build/v2";
+import { Inputs, v2, Result, FxError, Platform } from "@microsoft/teamsfx-api";
+import { FxCICDPluginResultFactory as ResultFactory } from "./result";
+import { CICDProviderFactory } from "./providers/factory";
+import { ProviderKind } from "./providers/enums";
+import { questionNames } from "./questions";
+import { InternalError, NoProjectOpenedError } from "./errors";
+import { telemetryHelper } from "./utils/telemetry-helper";
+import { Logger } from "./logger";
+
+export class CICDImpl {
+  public async addCICDWorkflows(
+    context: Context,
+    inputs: Inputs,
+    envInfo: v2.EnvInfoV2
+  ): Promise<Result<any, FxError>> {
+    // 1. Key inputs (envName, provider, template) x (hostingType, ).
+    if (!inputs.projectPath) {
+      throw new NoProjectOpenedError();
+    }
+    const projectPath = inputs.projectPath;
+    // By default(VSC), get env name from plugin's own `target-env` question.
+    let envName = inputs[questionNames.Environment];
+    // TODO: add support for VS/.Net Projects.
+    if (inputs.platform === Platform.CLI) {
+      // In CLI, get env name from the default `env` question.
+      envName = envInfo.envName;
+    }
+    const providerName = inputs[questionNames.Provider];
+    const templateNames = inputs[questionNames.Template] as string[];
+    if (!envName || !providerName || templateNames.length === 0) {
+      throw new InternalError("Some preconditions of inputs are not met.");
+    }
+
+    telemetryHelper.sendSuccessEvent(context, envInfo, "collect-cicd-questions", {
+      env: envName,
+      provider: providerName,
+      template: templateNames.join(","),
+    });
+
+    // 2. Call factory to get provider instance.
+    const providerInstance = CICDProviderFactory.create(providerName as ProviderKind);
+
+    // 3. Call instance.scaffold(template, replacements: any).
+    //  3.1 Call the initial scaffold.
+    const progressBar = context.userInteraction.createProgressBar(
+      "Scaffolding workflow automation files",
+      templateNames.length
+    );
+
+    const scaffoldedArr: boolean[] = [];
```

Comment:
Since we have for loop for templateNames[], can we also get the scaffolded array in the same loop? Something like this
```
const created: string[] = [];
const skipped: string[] = [];

let scaffolded = await providerInstance.scaffold(projectPath, templateNames[0], ...);
if (scaffolded.isOk() && scaffolded.value) {
  created.push(templateNames[0]);
} else {
  skipped.push(templateNames[0]);
}

...

for (const templateName of templateNames.slice(1)) {
  scaffolded = await providerInstance.scaffold(projectPath, templateName, ...);
  if (scaffolded.isOk() && scaffolded.value) {
    created.push(templateName);
  } else {
    skipped.push(templateName);
  }
}
```
So the step 4 for constructing message will be easy as 
```
let message = "";
if (created.length > 0) {
  message += ...;
}
if (skipped.length > 0) {
  message += ...;
}
``